jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java

```diff
@@ -291,10 +291,66 @@ void routeTargetPortFromServicePort() {
 
         // Then
         assertThat(route).isNotNull()
-                .extracting("metadata.name", "spec.host", "spec.to.kind", "spec.to.name", "spec.port.targetPort.intVal")
+                .extracting("metadata.name", "spec.host", "spec.to.kind", "spec.to.name", "spec.port.targetPort.value")
+                .contains("test-svc", "example.com", "Service", "test-svc", "http");
+    }
+
+    @Test
+    void routeTargetPortFromServicePort() {
+        // Given
+        ServiceBuilder serviceBuilder = new ServiceBuilder()
+                .editOrNewMetadata()
+                .withName("test-svc")
+                .addToLabels("expose", "true")
+                .endMetadata()
+                .editOrNewSpec()
+                .addNewPort()
+                .withPort(80)
+                .withProtocol("TCP")
+                .withTargetPort(new IntOrString(8080))
+                .endPort()
+                .addToSelector("group", "test")
+                .withType("LoadBalancer")
+                .editMatchingPort(e -> Boolean.TRUE).withPort(80).endPort().endSpec();
```

Comment:
Why is the Service being edited here to modify port with same value?

Reply:
okay, maybe you copied it from the existing test `portName_RouteTargetPortFromServicePort`.  When I removed it test still passes, could you please check if we can omit it?